cirq-ft/cirq_ft/algos/mean_estimation/arctan.py

```diff
@@ -31,18 +33,32 @@ class ArcTan(cirq.ArithmeticGate):
             in the output register of size `target_bitsize`.
     """
 
-    selection_bitsize: int
+    selection_bitsize_before_decimal: int
+    selection_bitsize_after_decimal: int
```

Comment:
I think we should create a new type that represents a quantum float. This type would be responsible to store `word_bitsize`  and `frac_bitsize` and would provide convenience methods like `.bitsize` which is a sum of `word_bitsize` and `frac_bitsize` and also bi-directional conversion to/from binary strings and floating point numbers. 

I imagine this would be useful in almost all arithmetic gates s.t. and not just `arctan` here. 

cc @mpharrigan In Qualtran, we have `bitsize: int` on `register` class that signifies the bit length of an integer. I think we should change this to a `dtype: QInt` by default, which would only need to store a bitsize like we do now,  and then start adding other abstractions like `QFloat` etc. where a single `bitsize: int` is not sufficient.